src/core/src/op/round.cpp

```diff
@@ -15,50 +16,48 @@ using namespace std;
 using namespace ngraph;
 
 namespace roundop {
-namespace {
-// function used by TYPE_CASE
-template <element::Type_t ET>
-inline bool evaluate(const HostTensorPtr& arg0,
-                     const HostTensorPtr& out,
-                     const size_t count,
-                     const op::v5::Round::RoundMode mode) {
-    using T = typename element_type_traits<ET>::value_type;
-    runtime::reference::round<T>(arg0->get_data_ptr<ET>(), out->get_data_ptr<ET>(), count, mode);
-    return true;
-}
 
-// function used by COPY_TENSOR
-template <element::Type_t ET>
-inline bool copy_tensor(const HostTensorPtr& arg0, const HostTensorPtr& out, const size_t count) {
-    runtime::reference::copy(arg0->get_data_ptr<ET>(), out->get_data_ptr<ET>(), count);
-    return true;
-}
+class Evaluate : public ov::element::NoAction<bool> {
+    template <element::Type_t ET>
+    static constexpr bool is_floating() {
+        return (ET == element::f16) || (ET == element::f32) || (ET == element::bf16);
+    }
 
+public:
+    using ov::element::NoAction<bool>::visit;
+    template <element::Type_t ET, typename std::enable_if<!is_floating<ET>()>::type* = nullptr>
+    static result_type visit(const HostTensorPtr& arg0,
+                             const HostTensorPtr& out,
+                             const size_t count,
+                             const op::v5::Round::RoundMode) {
+        memcpy(out->get_data_ptr(), arg0->get_data_ptr(), out->get_size_in_bytes());
```

Comment:
From my point of view will be better to have this differentiation inside `ngraph::runtime::reference::round()` function. Inside this function we need to select the optimal implementation.

Reply:
Agree, created ticket to refactor reference implementation and apply this change.